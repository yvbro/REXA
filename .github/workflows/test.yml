name: tests

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:
    runs-on: ubuntu-latest
    services:
      postgres:
        image: postgres:12-alpine
        env:
          POSTGRES_USER: rexaAdmin
          POSTGRES_PASSWORD: rexaAdmin
          POSTGRES_DB: rexa
          POSTGRES_HOST: localhost
        ports:
          - 5442:5432
        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5

    steps:
      - uses: actions/checkout@v2
      - name: Set up JDK 11
        uses: actions/setup-java@v1
        with:
          java-version: 11
      - name: Init db
        run: mvn flyway:migrate
      - name: Unit test and integration test
        run: mvn -B clean verify --file pom.xml