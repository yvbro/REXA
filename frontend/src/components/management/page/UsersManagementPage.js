import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { fetchUsers } from '../redux/userDuck';
import LoadingIndicator from '../../common/LoadingIndicator';
import UserListPage from '../smart/UserListPage';

const UsersManagementPage = () => {
    const dispatch = useDispatch();

    useEffect(() => {
        dispatch(fetchUsers());
    }, [dispatch]);

    const { loading } = useSelector((state) => ({
        loading: state.user.loading,
    }));

    return <>{loading ? <LoadingIndicator /> : <UserListPage />}</>;
};

export default UsersManagementPage;
