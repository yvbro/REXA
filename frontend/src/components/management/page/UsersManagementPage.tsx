import React, { useState } from 'react';
import { useQuery } from 'react-query';
import axios from 'axios';

import UserListPage from '../smart/UserListPage';
import { UserRexa } from '../../../models/management/UserRexa';
import { Page } from '../../../models/Page';
import LoadingIndicator from '../../common/LoadingIndicator';
import useUsersManagementService from '../../../services/useUsersManagementService';

const UsersManagementPage = () => {
    const [page, setPage] = useState(0);
    const [rowsPerPage, setRowsPerPage] = useState(10);

    const { fetchUsers } = useUsersManagementService();

    const {
        isLoading,
        isFetching,
        data: users,
    } = useQuery(['fetchUsers', page, rowsPerPage], () =>
        fetchUsers(page, rowsPerPage)
    );

    if (isLoading || isFetching) {
        return <LoadingIndicator />;
    }

    return (
        <UserListPage
            page={page}
            setPage={setPage}
            rowsPerPage={rowsPerPage}
            pageOfUsers={users?.data!!}
            setRowsPerPage={setRowsPerPage}
        />
    );
};

export default UsersManagementPage;
