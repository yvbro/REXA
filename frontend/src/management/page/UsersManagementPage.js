import React, { useEffect } from 'react';

import { useDispatch, useSelector } from 'react-redux';

import AppLayout from '../../app/AppLayout';
import { fetchUsers } from '../redux/userDuck';
import LoadingIndicator from '../../common/LoadingIndicator';
import UserListPage from '../smart/UserListPage';

const UsersManagementPage = () => {

    const dispatch = useDispatch();

    useEffect(() => {
        dispatch(fetchUsers());
    }, [dispatch]);

    const { loading } = useSelector((state) => ({
        loading: state.user.loading,
    }));

    return (
        <AppLayout>
            {loading ? <LoadingIndicator /> : <UserListPage />}
        </AppLayout>
    );
};

export default UsersManagementPage;
