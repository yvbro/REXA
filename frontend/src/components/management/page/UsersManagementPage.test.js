import React from 'react';
import {renderWithStore, cleanup} from '../../../helpers/test/test-utils';
import '@testing-library/jest-dom/extend-expect';

import UsersManagementPage from './UsersManagementPage';

const TEST_USERS = [
    { email: 'admin@test.com', roles: ['ADMIN', 'USER'], enabled: true},
    { email: 'user@test.com', roles: ['USER'], enabled: true},
    { email: 'disabled@test.com', roles: ['USER'], enabled: false},
];

const REGULAR_STATE = {
    user: {
        data: TEST_USERS,
        loading: false,
    }
};

describe('The UsersManagementPage component', () => {
    afterEach(cleanup);

    test('should take a snapshot', () => {
        const { asFragment } = renderWithStore(<UsersManagementPage />, REGULAR_STATE);

        expect(asFragment(<UsersManagementPage />)).toMatchSnapshot();
    });

    test('should display LoadingIndicator if loading', () => {
        const { getByRole } = renderWithStore(<UsersManagementPage />, {user: {loading: true}});

        expect(getByRole('progressbar')).toBeInTheDocument();
    });

    test('should display table if not loading', () => {
        const { getByRole } = renderWithStore(<UsersManagementPage />, REGULAR_STATE);

        expect(getByRole('heading')).toHaveTextContent('User Management');
        expect(getByRole('table')).toBeInTheDocument();
    });
});