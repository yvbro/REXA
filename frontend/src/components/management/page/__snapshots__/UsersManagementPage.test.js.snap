// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The UsersManagementPage component should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root rootDiv MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12"
    >
      <div
        class="MuiPaper-root MuiCard-root card MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <h5
            class="MuiTypography-root makeStyles-bold-2 MuiTypography-h5 MuiTypography-gutterBottom"
          >
            User Management
          </h5>
          <div
            class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiTableContainer-root makeStyles-root100-4"
            >
              <table
                aria-label="values table"
                class="MuiTable-root MuiTable-stickyHeader"
              >
                <thead
                  class="MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-6 MuiTableCell-alignCenter MuiTableCell-stickyHeader"
                      scope="col"
                    >
                      Email
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-6 MuiTableCell-alignCenter MuiTableCell-stickyHeader"
                      scope="col"
                    >
                      Role
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-6 MuiTableCell-alignCenter MuiTableCell-stickyHeader"
                      scope="col"
                    >
                      Enabled
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-6 MuiTableCell-alignCenter MuiTableCell-stickyHeader"
                      scope="col"
                    >
                      Actions
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-8"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-7 MuiTableCell-alignCenter"
                    >
                      admin@test.com
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-7 MuiTableCell-alignCenter"
                    >
                      ADMIN,USER
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-7 MuiTableCell-alignCenter"
                    >
                      <span
                        class="MuiSwitch-root"
                      >
                        <span
                          aria-disabled="true"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-checked-10 Mui-checked PrivateSwitchBase-disabled-11 Mui-disabled Mui-disabled Mui-disabled"
                          tabindex="-1"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <input
                              aria-label="primary checkbox"
                              checked=""
                              class="PrivateSwitchBase-input-12 MuiSwitch-input"
                              disabled=""
                              name="disableUser"
                              type="checkbox"
                              value=""
                            />
                            <span
                              class="MuiSwitch-thumb"
                            />
                          </span>
                        </span>
                        <span
                          class="MuiSwitch-track"
                        />
                      </span>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-7 MuiTableCell-alignCenter"
                    >
                      <span
                        aria-disabled="true"
                        aria-label="edit password"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary Mui-disabled Mui-disabled"
                        role="button"
                        tabindex="-1"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                            />
                          </svg>
                        </span>
                      </span>
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-8"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-7 MuiTableCell-alignCenter"
                    >
                      user@test.com
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-7 MuiTableCell-alignCenter"
                    >
                      USER
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-7 MuiTableCell-alignCenter"
                    >
                      <span
                        class="MuiSwitch-root"
                      >
                        <span
                          aria-disabled="false"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-checked-10 Mui-checked"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <input
                              aria-label="primary checkbox"
                              checked=""
                              class="PrivateSwitchBase-input-12 MuiSwitch-input"
                              name="disableUser"
                              type="checkbox"
                              value=""
                            />
                            <span
                              class="MuiSwitch-thumb"
                            />
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </span>
                        <span
                          class="MuiSwitch-track"
                        />
                      </span>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-7 MuiTableCell-alignCenter"
                    >
                      <span
                        aria-disabled="false"
                        aria-label="edit password"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                        role="button"
                        tabindex="0"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </span>
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-8"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-7 MuiTableCell-alignCenter"
                    >
                      disabled@test.com
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-7 MuiTableCell-alignCenter"
                    >
                      USER
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-7 MuiTableCell-alignCenter"
                    >
                      <span
                        class="MuiSwitch-root"
                      >
                        <span
                          aria-disabled="false"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiSwitch-switchBase MuiSwitch-colorPrimary"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <input
                              aria-label="primary checkbox"
                              class="PrivateSwitchBase-input-12 MuiSwitch-input"
                              name="disableUser"
                              type="checkbox"
                              value=""
                            />
                            <span
                              class="MuiSwitch-thumb"
                            />
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </span>
                        <span
                          class="MuiSwitch-track"
                        />
                      </span>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-7 MuiTableCell-alignCenter"
                    >
                      <span
                        aria-disabled="false"
                        aria-label="edit password"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                        role="button"
                        tabindex="0"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </span>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <table
              aria-label="pagination table"
              class="MuiTable-root"
            >
              <tfoot
                class="MuiTableFooter-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-footer"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-footer MuiTablePagination-root"
                    colspan="3"
                  >
                    <div
                      class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
                    >
                      <div
                        class="MuiTablePagination-spacer"
                      />
                      <p
                        class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
                        id="mui-1846"
                      >
                        Rows per page:
                      </p>
                      <div
                        class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
                      >
                        <div
                          aria-haspopup="listbox"
                          aria-labelledby="mui-1846 mui-17346"
                          class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
                          id="mui-17346"
                          role="button"
                          tabindex="0"
                        >
                          10
                        </div>
                        <input
                          aria-hidden="true"
                          class="MuiSelect-nativeInput"
                          tabindex="-1"
                          value="10"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M7 10l5 5 5-5z"
                          />
                        </svg>
                      </div>
                      <p
                        class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
                      >
                        1-3 of 3
                      </p>
                      <div
                        class="MuiTablePagination-actions"
                      >
                        <button
                          aria-label="Previous page"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                          disabled=""
                          tabindex="-1"
                          title="Previous page"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                              />
                            </svg>
                          </span>
                        </button>
                        <button
                          aria-label="Next page"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                          disabled=""
                          tabindex="-1"
                          title="Next page"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                              />
                            </svg>
                          </span>
                        </button>
                      </div>
                    </div>
                  </td>
                </tr>
              </tfoot>
            </table>
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Add user
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
